modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/ast/MacroExpansionModuleModel.java

```diff
@@ -137,6 +147,89 @@ public ArtifactAst expand() {
   }
 
   private ArtifactAst expand(List<ComponentAst> moduleComponentModels, Set<String> moduleGlobalElementsNames) {
+    if (existOperationThatUsesImplicitConfiguration() && hasXmlSdkPropertiesWithDefaultValues()) {
+
+      ComponentAst componentAst = new BaseComponentAst() {
+
+        @Override
+        public ComponentIdentifier getIdentifier() {
+          return ComponentIdentifier.builder()
+              .namespaceUri(extensionModel.getXmlDslModel().getNamespace())
+              .namespace(extensionModel.getXmlDslModel().getPrefix())
+              .name(IMPLICIT_CONFIG_NAME).build();
+        }
+
+        @Override
+        public ComponentType getComponentType() {
+          return CONFIG;
+        }
+
+        @Override
+        public ComponentLocation getLocation() {
+          return from(IMPLICIT_CONFIG_NAME);
+        }
+
+        @Override
+        public ComponentMetadataAst getMetadata() {
+          return EMPTY_METADATA;
+        }
+
+        @Override
+        public ComponentGenerationInformation getGenerationInformation() {
+          return EMPTY_GENERATION_INFO;
+        }
+
+        @Override
+        public Optional<String> getComponentId() {
+          return of(IMPLICIT_CONFIG_NAME);
+        }
+
+        @Override
+        public Map<String, Object> getAnnotations() {
+          return emptyMap();
+        }
+
+        @Override
+        public ExtensionModel getExtensionModel() {
+          return extensionModel;
+        }
+
+        @Override
+        public <M> Optional<M> getModel(Class<M> modelClass) {
+          return (Optional<M>) extensionModel.getConfigurationModel(MODULE_CONFIG_GLOBAL_ELEMENT_NAME)
+              .filter(modelClass::isInstance);
+        }
+
+        @Override
+        public MetadataType getType() {
+          return null;
+        }
+
+        @Override
+        public Collection<ComponentParameterAst> getParameters() {
+
+          final List<ParameterModel> parameterModels = getModel(ParameterizedModel.class)
```

Comment:
This should not be done every time the getParameters method is called. Either calculate it when the component is instantiated or the first time this is called only.